[armv7] mm/filemap.c:2532:23: error: ‘page’ undeclared (first use in this function)

On arm, gcc-5.3.1
```
  CC      mm/filemap.o
mm/filemap.c: In function ‘generic_file_buffered_read’:
mm/filemap.c:2532:23: error: ‘page’ undeclared (first use in this function)
     flush_dcache_page(page);
                       ^
mm/filemap.c:2532:23: note: each undeclared identifier is reported only once for each function it appears in
scripts/Makefile.build:275: recipe for target 'mm/filemap.o' failed
```

Same on arch=powerpc
`mm/filemap.c:2532:23: error: ‘page’ undeclared (first use in this function)  `

Compiler: gcc (Ubuntu 9.1.0-2ubuntu2~16.04) 9.1.0, Linux/powerpc 5.1.0

--- a/mm/filemap.c
+++ b/mm/filemap.c
@@ -2466,7 +2466,7 @@ static ssize_t generic_file_buffered_read(struct kiocb *iocb,
 			 * before reading the page on the kernel side.
 			 */
 			if (writably_mapped)
-				flush_dcache_page(page);
+				flush_dcache_page(pages[i]);
 
 			copied = copy_page_to_iter(pages[i], offset, bytes, iter);
 

